src/ast/diagnostic.h

```diff
@@ -0,0 +1,122 @@
+#pragma once
+
+#include <cassert>
+#include <functional>
+#include <memory>
+#include <sstream>
+#include <string>
+#include <variant>
+#include <vector>
+
+#include "location.hh"
+
+namespace bpftrace {
+namespace ast {
+
+// Diagnostic reflects a single error at a single source location. This is a
+// simple wrapper around a string for that message, and the location class.
+class Diagnostic {
+public:
+  enum class Severity {
+    Warning,
+    Error,
```

Comment:
Do we also have (or want to have) `Info` severity?

Reply:
We don't have `LOG(INFO)` at the moment but instead `LOG(V1)`. I don't think we need either location or hints for these types of messages.